### ISSUE TITLE
SSC-026: ClassifyTrainer: production training loop with validation/checkpointing/LR schedule

### ISSUE THREAD

**Issue by noahgift** [paiml/entrenar#96]
## SSC-026: Production Training Loop

**Priority**: P1
**Blocked by**: SSC-023 (tokenizer), SSC-025 (batch training)
**Blocks**: SSC-027 (CLI execution)
**Contract**: `provable-contracts/contracts/aprender/training-loop-v1.yaml`

### Problem

No complete training loop with epoch management, validation split, metrics, checkpointing, and LR scheduling. The current code only has `train_step()` / `train_batch()` — individual steps without orchestration.

### What Exists

- LR schedulers in aprender: CosineAnnealing, LinearWarmup, StepDecay, etc.
- Checkpoint infrastructure in entrenar
- Loss/accuracy computation in ClassifyPipeline
- `train_step()` and (after SSC-025) `train_batch()`

### What's Missing

**`ClassifyTrainer`** struct that orchestrates:

```rust
pub struct ClassifyTrainer {
    pipeline: ClassifyPipeline,
    config: TrainingConfig,
    lr_scheduler: Box<dyn LrScheduler>,
    best_val_accuracy: f32,
    best_val_loss: f32,
    epoch_metrics: Vec<EpochMetrics>,
}

pub struct TrainingConfig {
    pub epochs: usize,           // default: 50
    pub batch_size: usize,       // default: 32
    pub val_split: f32,          // default: 0.2
    pub lr: f32,                 // default: 2e-5
    pub warmup_steps: usize,     // default: 100
    pub save_every: usize,       // default: 5 epochs
    pub checkpoint_dir: PathBuf,
    pub early_stopping_patience: Option<usize>,
}
```

**Training loop**:
```
for epoch in 0..config.epochs {
    1. Shuffle training data
    2. Split into batches
    3. For each batch: train_batch() → accumulate metrics
    4. Compute validation metrics (accuracy, loss, per-class F1)
    5. LR scheduler step
    6. Log metrics (epoch, train_loss, val_loss, val_acc, lr)
    7. Checkpoint if val_loss improved
    8. Early stopping check
}
```

**Wired to CLI**: `apr finetune --task classify --data corpus.jsonl` invokes `ClassifyTrainer::train()`

### Contract Invariants

| ID | Invariant | Description |
|----|-----------|-------------|
| F-LOOP-001 | Loss decreasing | EMA(loss, window=5) is decreasing over training |
| F-LOOP-002 | Validation computed | Validation accuracy computed every epoch |
| F-LOOP-003 | Checkpoint restorable | Loading checkpoint restores model to same val_loss ± ε |
| F-LOOP-004 | LR schedule applied | Learning rate changes according to scheduler |
| F-LOOP-005 | Data shuffled | Training order differs between epochs (seeded RNG) |

### Acceptance Criteria

- [ ] `ClassifyTrainer::train(data)` runs complete training loop
- [ ] Train/val split with configurable ratio
- [ ] Per-epoch metrics: train_loss, val_loss, val_accuracy, learning_rate
- [ ] Checkpoint saving on best val_loss
- [ ] Checkpoint loading resumes training
- [ ] LR scheduling (warmup + cosine annealing)
- [ ] Early stopping (optional, patience-based)
- [ ] Convergence on 15-sample demo corpus (val_acc > 60%)
- [ ] All contract invariants verified
- [ ] Integration test: 26K samples × 5 epochs completes in <10min

### References

- LR schedulers in `aprender/src/nn/optim/scheduler.rs`
- Spec: `bashrs/docs/specifications/shell-safety-inference.md` Section 14
